src/daemon/rpmostreed-sysroot.c

```diff
@@ -727,6 +727,11 @@ rpmostreed_sysroot_populate (RpmostreedSysroot *self,
   if (!ostree_sysroot_get_repo (self->ot_sysroot, &self->repo, cancellable, error))
     return FALSE;
 
+  /* Migrate legacy pkgcache repo into system repo. After the first time, this boils down to
+   * one stat() call. */
+  if (!rpmostree_migrate_pkgcache_repo (self->repo, cancellable, error))
```

Comment:
This is a potentially expensive operation to do while blocking the main loop on startup.  Feels like we should model this as a transaction?

Reply:
(In practice for me at least my main system is NVMe and I barely notice IO but still I think the separate txn is the right thing to do)

Reply:
I actually had this be part of the deploy transaction before. I switched to occur at daemon startup since even large numbers of pkgs didn't incur a noticeable delay. Though yeah, that's a skewed perspective here too since I have fast storage. :)

Giving it its own transaction is an interesting idea. Though since it shouldn't be very long-lived, it seems like a bit overkill vs e.g. sneaking it into the deploy transaction?

Reply:
Ah, doing it on-demand in deploy seems nicer to me indeed; that way we only change things if the admin changes the current deployment after booting too, so the "update, find breakage, rollback" scenario will still be OK right?